### PR TITLE
Network protocol for Python

### PR DESCRIPTION
**Pull request by saileshmittal**
This is same as https://github.com/twitter/heron/pull/658 but without eggs.

### CODE REVIEW

**Comment by kramasamy** [comment]
@saileshmittal - have all the review comments taken care of.

**Comment by saileshmittal** [comment]
@maosongfu , @billonahill could you take a look at this PR?

**Comment by billonahill** [comment]
The python BUILD parts that I commented on in #658 have been addressed. I have not reviewed the code itself.

File: heron/common/src/python/network/packet.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  def unpackProtobuf(self, message):
+    """
+    Deserialize the protobuf message
+    """
+    messageSize = self.unpackInt(self.position)
+    self.position += self.INT_SIZE_IN_BYTES
+    message.ParseFromString(self.data[self.position : self.position + messageSize])
+    return message
+
+def pack(reqId, message):
```
Comment (maosongfu) [inline heron/common/src/python/network/packet.py]: Can u make it into two classes: IncomingPacket & OutgoingPacket with similar interfaces in c++/java IncomingPacket & OutgoingPacket?

Comment (kramasamy) [inline heron/common/src/python/network/packet.py]: @saileshmittal - can you incorporate this?

**Comment by objmagic** [comment]
Should we close since @taishi8117 is working on similar task, which does not use the code in this PR?

**Comment by kramasamy** [comment]
closing this - since @taishi8117 finished it.

**Pull request closed without merging**